### PR TITLE
Wrapped view calls in exception handling.

### PR DESCRIPTION
**Pull request by michaek**
## Description
This prevents errors in view rendering from halting application execution. No error handling functionality is exposed to the application. 

## Motivation and Context
Following from https://github.com/MithrilJS/mithril.js/issues/1937
Errors in views for any component can currently be fatal to the entire application. That's unacceptable.

## How Has This Been Tested?
I've tested it against a fiddle provided by the original reporter of 1937 and against the application I'm working on. I haven't run mithril's ospec.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated `docs/change-log.md`

### CODE REVIEW

**Comment by michaek** [comment]
I've now run mithril's tests, and five render tests related to downstream error handling are failing. It's clear this is a breaking change for some cases, but the current behavior is significantly broken as well. There's tension in the API between handling thrown errors from calling render, and not letting thrown errors from calling redraw halt a mounted app (where you have no ability as a consumer to handle errors). There may be a middle ground that satisfies both.

File: mithril.js [hunk 1]
```diff
@@ -691,7 +704,11 @@ var coreRenderer = function($window) {
 		if (recycling) {
 			initComponent(vnode, hooks)
 		} else {
-			vnode.instance = Vnode.normalize(vnode._state.view.call(vnode.state, vnode))
+			try {
+				vnode.instance = Vnode.normalize(vnode._state.view.call(vnode.state, vnode))
+			} catch (err) {
+				console.error(err)
```
Comment (isiahmeadows) [inline mithril.js]: Please reset `vnode.instance` here to `undefined`, since it starts out with the old tree + state.

File: mithril.js [hunk 2]
```diff
@@ -397,6 +397,15 @@ var coreRenderer = function($window) {
...
 	}
+	// IE9 - IE11 (at least) throw an UnspecifiedError when accessing document.activeElement when
+	// inside an iframe. Catch and swallow this error0, and heavy-handidly return null.
+	function activeElement() {
+		try {
+			return $doc.activeElement
+		} catch (e) {
+			return null
+		}
+	}
```
Comment (isiahmeadows) [inline mithril.js]: Please open a separate PR for this. It doesn't need bundled with component error handling.

**Comment by barneycarroll** [comment]
@michaek could you produce a simple problem case to illustrate the change or describe the new failure case? I've always been appreciative of Mithril's "if it breaks, it breaks" with vanilla stack traces and sundry compared to other frameworks opinionated error catchment policies.

**Comment by isiahmeadows** [comment]
@barneycarroll I'd be okay with an optional per-component "catch", though, as I mentioned in #1937 itself. But as for this, I'm not a fan of silently swallowing uncaught errors like what this PR does.

@michaek Let's keep this discussion in #1937 for now, and then create a PR once we know what we want.

**Pull request closed without merging**